zerver/tests/test_events.py

```diff
@@ -2115,6 +2114,39 @@ def test_rename_stream(self) -> None:
         ])
         error = schema_checker('events[1]', events[1])
         self.assert_on_error(error)
+        schema_checker = check_dict([
+            ('stream_email_notify', equals(False)),
+            ('flags', check_list(check_string)),
+            ('email_notified', equals(True)),
+            ('type', equals('message')),
+            ('message', check_dict([
+                ('timestamp', check_int),
+                ('content', equals(notification)),
+                ('content_type', equals('text/html')),
+                ('sender_email', equals('notification-bot@zulip.com')),
+                ('sender_id', check_int),
+                ('sender_short_name', equals('notification-bot')),
+                ('display_recipient', equals(new_name)),
+                ('id', check_int),
+                ('stream_id', check_int),
+                ('sender_realm_str', check_string),
+                ('sender_full_name', equals('Notification Bot')),
+                ('is_me_message', equals(False)),
+                ('type', equals('stream')),
+                ('submessages', check_list(check_string)),
+                (TOPIC_LINKS, check_list(check_url)),
+                ('avatar_url', check_url),
+                ('reactions', check_list(None)),
+                ('client', equals('Internal')),
+                (TOPIC_NAME, equals('welcome')),
+                ('recipient_id', check_int)
```

Comment:
We probably don't need to check all of these fields, since this test is not really about verifying the format of messages.